experimental/dds/tree2/src/test/shared-tree/sharedTreeView.spec.ts

```diff
@@ -0,0 +1,70 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { strict as assert } from "assert";
+import { SchemaBuilder, Any } from "../../feature-libraries";
+import { createSharedTreeView } from "../../shared-tree";
+import { ValueSchema, AllowedUpdateType, storedEmptyFieldSchema } from "../../core";
+
+const builder = new SchemaBuilder("Schematize Tree Tests");
+const root = builder.leaf("root", ValueSchema.Number);
+const schema = builder.intoDocumentSchema(SchemaBuilder.fieldOptional(Any));
+
+const builderGeneralized = new SchemaBuilder("Schematize Tree Tests Generalized");
+const rootGeneralized = builderGeneralized.leaf("root", ValueSchema.Serializable);
+const schemaGeneralized = builderGeneralized.intoDocumentSchema(SchemaBuilder.fieldOptional(Any));
+
+describe("sharedTreeView", () => {
+	describe("schematize", () => {
+		it("initialize tree", () => {
+			const tree = createSharedTreeView();
+			assert.equal(tree.storedSchema.rootFieldSchema, storedEmptyFieldSchema);
+
+			tree.schematize({
+				allowedSchemaModifications: AllowedUpdateType.None,
+				initialTree: 10 as any,
+				schema,
+			});
+			assert.equal(tree.root, 10);
+		});
+
+		it("noop upgrade", () => {
+			const tree = createSharedTreeView();
+			tree.storedSchema.update(schema);
+
+			// No op upgrade with AllowedUpdateType.None does not error
+			const schematized = tree.schematize({
+				allowedSchemaModifications: AllowedUpdateType.None,
+				initialTree: 10,
+				schema,
+			});
+			// And does not add initial tree:
+			assert.equal(schematized.root, undefined);
+		});
```

Comment:
This scenario confuses me a bit. Why is the schematize call a noop? Because update was called first? I would expect the update to be a noop but the schematize to work.

Reply:
this test checks that the schematize that should produce no ops works (even when we require it to not edit anything). I think thats what you are saying to expect, and what it does.